fix: content now prioritizes doc.state.next instead of doc.state

isidorosp opened the pull request:
@oaksprout can you check if this fixes the issue where sometimes we weren't pulling the correct document state on your end?

isidorosp committed:
fix: content now prioritizes doc.state.next instead of doc.state

oaksprout reviewed (commented):
When I tested this it looked promising at first, but there still seems to be an issue when switching between commits. I recorded a video to show the issue:

https://user-images.githubusercontent.com/66292936/116101293-9f752000-a6a5-11eb-8b7e-31d4b054f7d7.mov

isidorosp committed:
fixes

TODO: Fix bug where if you browse to a commit, then go to a new document, the document fails to load unless you refresh

oaksprout reviewed (approved):
LGTM!

I like the visual update on the loading, too – feels much cleaner and more consistent now 🙂
